apps/web/components/ui/write-review-form.tsx

```diff
@@ -50,23 +52,64 @@ export default function WriteReviewForm({ onSubmit, previousReview }: WriteRevie
 		content: previousReview?.description || ""
 	});
 
+	//to save localStorage when state changes
+	useEffect(() => {
+		if (!previousReview) {
+			const savedAcademicYear = localStorage.getItem(`reviewFormAcademicYear_${courseCode}`);
+			const savedRating = localStorage.getItem(`reviewFormRating_${courseCode}`);
+			const savedDescription = localStorage.getItem(`reviewFormDescription_${courseCode}`);
+
+			if (savedAcademicYear) setAcademicYear(savedAcademicYear);
+			if (savedRating) setRating(parseFloat(savedRating));
+			if (savedDescription && markdownEditor) markdownEditor.commands.setContent(savedDescription);
+		}
+	}, [previousReview, markdownEditor]);
+
+	
+	useEffect(() => {
+		localStorage.setItem(`reviewFormRating_${courseCode}`, rating.toString());
+	}, [rating]);
+
+	useEffect(() => {
+		if (academicYear) localStorage.setItem(`reviewFormAcademicYear_${courseCode}`, academicYear);
+		if (markdownEditor) {
+			const saveMarkdown = () => {
+				localStorage.setItem(`reviewFormDescription_${courseCode}`, markdownEditor.storage.markdown.getMarkdown());
+			};
+			markdownEditor.on("update", saveMarkdown);
+			return () => {
+				markdownEditor.off("update", saveMarkdown);
+			};
+		}
+	}, [markdownEditor, academicYear, ]);
+
+	const handleSubmit = (e: React.FormEvent) => {
+		e.preventDefault();
+		const currentDescription = markdownEditor?.storage.markdown.getMarkdown() || "";
+		if (!academicYear || !currentDescription || !rating) {
+			notifications.show({
+				title: "Hold on! Your review still contains some missing fields",
+				color: "red",
+				message:
+					"Make sure you have filled all the fields such as academic year, ratings, and review descriptions",
+				autoClose: 5000
+			});
+			return;
+		}
+
+		onSubmit(academicYear, currentDescription, rating);
+
+		// to remove the local storage
+		localStorage.removeItem(`reviewFormAcademicYear_${	courseCode}`);
+		localStorage.removeItem(`reviewFormRating_${courseCode}`);
+		localStorage.removeItem(`reviewFormDescription${courseCode}`);
```

Comment:
hold on, you're already deleting this in the onSubmit call back right? Then why are we deleting again in `[courseCode]/page.tsx`? And... why does this seem like it will always delete from the local storage regardless of the submission result

Reply:
Also, the keys are not matching when you set, get, and remove for review description

You set and get with `reviewFormDescription_${courseCode}` but remove with `reviewFormDescription_${courseCode}`